QA: Introduce feature_blocksign.py

jtimon opened the pull request:
Let's add this test that @stevenroose wrote, I only slightly modified it.

Later I want to extend it to test things other than 3 of 3, including a single signer (without using multisig if it's 1 of 1).

instagibbs commented on qa/rpc-tests/feature_blocksign.py:
```diff
@@ -0,0 +1,114 @@
+            self.sync_all()
+
+    def run_test(self):
+        # Check that there's 100 UTXOs on each of the nodes
```
I don't think these checks are useful in this test. Check the block height instead?

instagibbs commented on qa/rpc-tests/feature_blocksign.py:
```diff
@@ -0,0 +1,114 @@
+    def __init__(self):
+        super().__init__()
+        self.setup_clean_chain = True
+        self.num_nodes = 3
```
can we factor out "3" for all these places in case we want to expand it?

jtimon commented:
Fixed nits, moved from 3-of-3 to 7-of-9, that makes it slower (from 11s to 34s in my hardware), but I think that's ok. We can expand further afterwards, but this works now.

instagibbs commented on qa/rpc-tests/feature_blocksign.py:
```diff
@@ -0,0 +1,117 @@
+
+        # collect required_signers signatures
+        sigs = []
+        for i in range(self.required_signers):
```
We should loop and make sure that combining below-threshhold doesn't give you a block considered valid.

At each inner-section we should combine and check that it's not complete, and cannot move the blockchain forward.

stevenroose commented:
Hehe, nice to see that thing being even the slightest bit inspiring :) It was mostly me getting familiar with the Python framework :)

Can't wait to join you guys!

instagibbs commented on qa/rpc-tests/feature_blocksign.py:
```diff
@@ -0,0 +1,124 @@
+        sigs = []
+        for i in range(self.required_signers):
+            result = miner.combineblocksigs(block, sigs)
+            util.assert_equal(result["complete"], False)
```
Please also do a submitblock, then check that it didn't move the chain forward. Belts and suspenders!

instagibbs commented on qa/rpc-tests/feature_blocksign.py:
```diff
@@ -0,0 +1,124 @@
+#!/usr/bin/env python3
+# requires Python 3.5
```
We shouldn't require a specific version of python3.

instagibbs commented:
utACK https://github.com/ElementsProject/elements/pull/373/commits/f66e94a7120caa4a9a53395853bc837d75aaed6b

@stevenroose See you soon!
